sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -0,0 +1,524 @@
+if (isNode) {
+  keyto = require("@trust/keyto");
+}
+
+/**
+ * The client to interact with the KeyVault cryptography functionality
+ */
+export class CryptographyClient {
+
+  /**
+   * Retrieves the complete key from the key vault
+   * @param options Options for retrieving key
+   */
+  public async getKey(options?: GetKeyOptions): Promise<JsonWebKey> {
+    if (typeof this.key === "string") {
+      if (!this.name || this.name === "") {
+        throw new Error("getKey requires a key with a name");
+      }
+      const key = await this.client.getKey(
+        this.vaultBaseUrl,
+        this.name,
+        options && options.version ? options.version : (this.version ? this.version : ""),
+        options
+      );
+      return key.key!;
+    } else {
+      return this.key;
+    }
+  }
+
+  /**
+   * Encrypts the given plaintext with the specified cryptography algorithm
+   * @param plaintext The text to encrypt
+   * @param algorithm The algorithm to use
+   * @param options Additional options
+   */
+  public async encrypt(
+    plaintext: Uint8Array,
+    algorithm: JsonWebKeyEncryptionAlgorithm,
+    options?: EncryptOptions
+  ): Promise<Uint8Array> {
+    if (isNode) {
+      await this.fetchFullKeyIfPossible();
+
+      if (typeof this.key !== "string") {
+        switch (algorithm) {
+          case "RSA1_5": {
+            let keyPEM = keyto.from(this.key, "jwk").toString('pem', 'public_pkcs1');
+
+            let padded: any = { key: keyPEM, padding: constants.RSA_PKCS1_PADDING };
+            const encrypted = crypto.publicEncrypt(padded, Buffer.from(plaintext));
+            return encrypted;
+          };
+          case "RSA-OAEP": {
+            let keyPEM = keyto.from(this.key, "jwk").toString('pem', 'public_pkcs1');
+
+            const encrypted = crypto.publicEncrypt(keyPEM, Buffer.from(plaintext));
+            return encrypted;
+          };
+        }
+      }
+    }
+
+    // Default to the service
+    let result = await this.client.encrypt(this.vaultBaseUrl, this.name, this.version, algorithm, plaintext, options);
+    return result.result!;
+  }
+
+  /**
+   * Decrypts the given ciphertext with the specified cryptography algorithm
+   * @param ciphertext The ciphertext to decrypt
+   * @param algorithm The algorithm to use
+   * @param options Additional options
+   */  
+  public async decrypt(
+    ciphertext: Uint8Array,
+    algorithm: JsonWebKeyEncryptionAlgorithm,
+    options?: DecryptOptions
+  ): Promise<Uint8Array> {
+    let result = await this.client.decrypt(this.vaultBaseUrl, this.name, this.version, algorithm, ciphertext, options);
+    return result.result!;
+  }
+
+  /**
+   * Wraps the given key using the specified cryptography algorithm
+   * @param key The key to wrap
+   * @param algorithm The encryption algorithm to use to wrap the given key
+   * @param options Additional options
+   */
+  public async wrapKey(
+    key: Uint8Array,
+    algorithm: JsonWebKeyEncryptionAlgorithm,
```

Comment:
>JsonWebKeyEncryptionAlgorithm [](start = 15, length = 29)

This needs to be it's own "enum" KeyWrapAlgorithm, which has the RSA encryption algorithms, but eventually will contain more.